File: Modules/UserContext/UserContext.Core/ValueObject/Location.cs
namespace UserContext.Core.ValueObject;

public record Location
{
    public string Country { get; private set; } = default!;
    public string City { get; private set; } = default!;
    public string State { get; private set; } = default!;
    public string PostalCode { get; private set; } = default!;
    public string? Steet {get;private set;} = default!;
    public string? StreetNumber {get; private set;} =default!;
    private Location() { }
    public static Location Create(
        string country,
        string city,
        string state,
        string postal
    )
    {
        Location newLocation = new();
        newLocation.Country = country.Trim();
        newLocation.City = city.Trim();
        newLocation.State = state.Trim();
        newLocation.PostalCode = postal.Trim();
        return newLocation;
    }
    
    public static Location Create(
        string country,
        string city,
        string state,
        string postal,
        string street,
        string streetNumber
    )
    {
        Location newLocation = new();
        newLocation.Country = country.Trim();
        newLocation.City = city.Trim();
        newLocation.State = state.Trim();
        newLocation.PostalCode = postal.Trim();
        newLocation.Steet = street.Trim();
        newLocation.StreetNumber = streetNumber.Trim();
        return newLocation;
    }

}

File: client/src/pages/api/user/location.ts
import { getSession } from "@auth0/nextjs-auth0";
import axios from "axios";
import { NextApiRequest, NextApiResponse } from "next";

async function handler(req:NextApiRequest,res:NextApiResponse)
{
    const session = await getSession(req,res);
    if(session == null || session == undefined){
        res.status(401).json({ok:false,message:"Unathorized"})
        return;
    }
    if(session)
    {
        try{
            const data = await axios({
                url:`${process.env.API_URL}/users/${req.body.userId}/location`,
                method:"PATCH",
                headers:{
                    Accept:"application/json",
                    Authorization:`Bearer ${session.accessToken}`
                },
                data:{
                    country:req.body.country,
                    city:req.body.city,
                    state:req.body.state,
                    postalCode:req.body.postalCode,
                    street:req.body.street,
                    streetNumber:req.body.streetNumber
                }
            })
            if(data.status == 200)
            {
                res.status(200).json({ok:true,message:"Profile modified"})
                return;
            }
        }catch(e)
        {
            res.status(500).json({ok:false,message:"Error on Api server"})
            return;
        }
    }
    res.status(400).json({ok:false,message:"Unathorized"})
    return;
}

export default handler